src/main/java/it/smartcommunitylab/aac/webauthn/provider/WebAuthnAccountService.java

```diff
@@ -0,0 +1,242 @@
+            String subject = parseResourceId(attributes.get("userId"));
+            account = userAccountService.findByProviderAndSubject(provider, subject);
+        }
+
+        if (account == null
+                && attributes.keySet().containsAll(Arrays.asList("realm", "username"))
+                && provider.equals((attributes.get("provider")))) {
+            account = userAccountService
+                    .findByProviderAndUsername(provider, attributes.get("username"));
+        }
+
+        if (account == null
+                && attributes.keySet().contains("credentialId")) {
+            try {
+                account = userAccountService.findByCredentialId(
+                        ByteArray.fromBase64Url(attributes.get("credentialId")));
+            } catch (Exception e) {
+            }
+        }
+
+        if (account == null) {
+            throw new NoSuchUserException("No WebAuthn user found matching attributes");
+        }
+
+        String username = account.getUsername();
+
+        // set providerId since all WebAuthn accounts have the same
+        account.setProvider(getProvider());
+
+        // rewrite webauthn userId
+        account.setSubject(exportInternalId(username));
+
+        return account;
+    }
+
+    @Override
+    public void deleteAccount(String subject) throws NoSuchUserException {
+        userAccountService.deleteAccount(getProvider(), subject);
+    }
+
+    @Override
+    public boolean canRegister() {
+        return config.isEnableRegistration();
+    }
+
+    @Override
+    public boolean canUpdate() {
+        return config.isEnableUpdate();
+    }
+
+    @Override
+    public WebAuthnUserAccount registerAccount(String subjectId, UserAccount reg)
+            throws NoSuchUserException, RegistrationException {
+        if (!config.isEnableRegistration()) {
+            throw new IllegalArgumentException("delete is disabled for this provider");
+        }
+
+        // we expect subject to be valid
+        if (!StringUtils.hasText(subjectId)) {
+            throw new RegistrationException("missing-subject");
+
+        }
+
+        String provider = getProvider();
+
+        // extract base fields
+        String username = Jsoup.clean(reg.getUsername(), Safelist.none());
+
+        // validate username
+        if (!StringUtils.hasText(username)) {
+            throw new RegistrationException("missing-username");
+        }
+
+        WebAuthnUserAccount account = userAccountService
+                .findByProviderAndUsername(provider, username);
+        if (account != null) {
+            throw new AlreadyRegisteredException("duplicate-registration");
+        }
+
+        // check type and extract our parameters if present
+        Set<WebAuthnCredential> credentials = null;
+        String email = null;
+
+        if (reg instanceof WebAuthnUserAccount) {
+            WebAuthnUserAccount ireg = (WebAuthnUserAccount) reg;
+            credentials = ireg.getCredentials();
```

Comment:
if you expect credentials in useraccount add a transient field to the model

Reply:
Addressed in ef11f270f4c510ec299cbf24206bf3a88ba45c4d